app/components/channel_list/categories/categories.tsx

```diff
@@ -22,26 +24,31 @@ const styles = StyleSheet.create({
 
 const extractKey = (item: CategoryModel) => item.id;
 
-const renderCategory = (data: {item: CategoryModel}) => {
-    return (
-        <>
-            <CategoryHeader category={data.item}/>
-            <CategoryBody category={data.item}/>
-        </>
-    );
-};
+const Categories = ({categories, currentChannelId}: Props) => {
+    const intl = useIntl();
+    const renderCategory = useCallback((data: {item: CategoryModel}) => {
+        return (
+            <>
+                <CategoryHeader category={data.item}/>
+                <CategoryBody
+                    category={data.item}
+                    currentChannelId={currentChannelId}
+                    locale={intl.locale}
+                />
+            </>
+        );
+    }, [categories, currentChannelId, intl.locale]);
 
-const Categories = (props: Props) => {
-    if (!props.categories.length) {
+    // Sort Categories
+    categories.sort((a, b) => a.sortOrder - b.sortOrder);
```

Comment:
I know the PR has already been merged.  But the `categories`, could they be sorted within the `withObservables` block under   `app/components/channel_list/categories/index.ts`  ? That would avoid sorting it each time the component renders?  @shazm

Reply:
The thing is that you want them to be sorted each time the component re-renders